install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -11,144 +11,406 @@
+other supported types. Note: the range of allowed ids can be defined to include any user IDs required for the target storage.
+<8> when set to _MustRunAs_, the container is created with the SCC's selinux options, or the
+MCS default defined in the namespace. A type of _RunAsAny_ indicates that selinux context is not
+required, and if not defined in the pod, is not set in the container.
+<9> The selinux user name, role name, type, and labels can be defined here.
+
+[[supplemental-groups]]
+== Supplemental Groups
+*Note:* the link:#scc[SCC overview], above, should be read before working with supplemental groups.
+
+*Note:* it's generally preferable to use group ids (supplemental or 
+link:#fsgroup[fsgroup]) to gain access to persistent storage vs. using 
+link:#user-id[user ids].
+
+Supplemental groups are regular Linux groups. When a process runs in Linux, it has a UID,
+a GID, and one or more supplemental groups. These attributes can be set for a container's
+main process. The `supplementalGroups` ids are typically used for controlling access to
+_shared_ storage, such as NFS and GlusterFS; whereas, link:#fsgroup[fsGroup] is used for
+controlling access to _block_ storage, such as Ceph-RBD and iSCSI.
+
+The Openshift _shared_ storage plug-ins mount volumes such that the POSIX
+permissions on the mount match the permissions on the target storage. For example, if the 
+target storage's owner id is 1234 and its group id is 5678, then the mount on
+the host node and in the container will have those same ids. Therefore, the
+container's main process  must match one or both of those ids in order to access the volume. 
+
+[[nfs-example]]
 For example, consider the following NFS export:
-
 ====
 ----
-showmount -e example.com
-Export list for example.com:
-/nfsexport (everyone)
-
-# ls -ld /nfsexport # on NFS server
-drwxrwx---. 2 root 1234 4096 Oct 30 15:27 /nfsexport
+#on an openshift node:
+#(Note: showmount needs access to the ports used by rpcbind and rpc.mount on the nfs server)
+showmount -e <nfs-server-ip-or-hostname>
+Export list for f21-nfs.vm:
+/opt/nfs  *
+
+#on the nfs server:
+# cat /etc/exports
+/opt/nfs *(rw,sync,no_root_squash)
+...
+
+# ls -lZ /opt/nfs -d
+drwxrws---. nobody 5555 unconfined_u:object_r:usr_t:s0   /opt/nfs
+
+# id nobody
+uid=99(nobody) gid=99(nobody) groups=99(nobody)
 ----
 ====
+*Note:* in this case the owner is 99 (nobody), but the suggestions and examples in this document apply to any non-root owner.
 
-The above export is only accessible by *root* and the group *1234*. So any pod
-which is not run as *root* and is not a member the group *1234* will not be able
-to access that export. It is often the case that the primary UID in the
-container cannot be controlled or predicted.
-
-To solve this issue, you can add the group *1234* to `*SupplementalGroups*` as
-in the following example:
-
-====
+The _/opt/nfs/_ export is accessible by UID *99* and the group *5555*. In general, containers
```

Comment:
That's true and I forget what the wackiness was (maybe related to container's UID?). That is one reason for the couple of notes about UID 99 not being a particularly special id. `nfsnobody` and `nobody` seem like reasonable ids to use in these examples since that is not an unreasonable id to have on an export. Wrong?